### ISSUE TITLE
Question/request: Linking to re-exported modules

### ISSUE THREAD

**Issue by chris-martin** [haskell/haddock#1353]
I recently put together [a package](https://hackage.haskell.org/package/sockets-and-pipes-0.1) that includes a lot modules re-exported from other packages, and the Cabal description includes links to these modules using haddock's double-quote syntax. I thought this was going well, because these links worked when I ran `cabal haddock` to generate the HTML documentation on my local machine. But when I uploaded to Hackage (`cabal sdist` and `cabal upload --publish`), I found that these links all 404. Is this a problem with Haddock, with Cabal, or with my own configuration?

**Comment by Kleidukos** [comment]
@chris-martin Indeed, it would seem that you've been bitten by the fact that link resolution is disabled when using the double commas.
![Screenshot from 2021-02-25 12-06-21](https://user-images.githubusercontent.com/29253044/109144783-ea37f180-7761-11eb-80e6-b34f722297ac.png)

**Comment by alexbiehl** [comment]
@chris-martin Thanks for reaching out. 

So from what I can tell so far is that: 

- As you say, I can confirm running locally works just fine 
- My initial guess was that Hackage-Builder wouldn't pass the Haddock interfaces to Haddock but since the re-exported modules are listed in the module tree it looks like it's being passed correctly. 
- Another guess. was that maybe the packages you linked had all their docs uploaded by user. It's known that linking is difficult in that case. Unfortunately turns out to be false too. 

Investigating further.

**Comment by alexbiehl** [comment]
Ok! Here we go: 

The package overview page on Hackage is rendered by Hackage and is using a different logic than the index page generated by Haddock!

See https://github.com/haskell/hackage-server/blob/bca51b91ffe592f50680df098dd9e1a0c75cda9f/Distribution/Server/Pages/Package.hs#L154-L157 for how Hackage resolves a module URL (surpise: it really just replaces `.` with `-` and appends `.html`). It simply lacks the capability to link to a module from another package. :(  

I am thinking of a way to unify the Hackage with Haddock but please don't hold your breath.

**Comment by alexbiehl** [comment]
One solution could be to have Haddock generate a hackage.json file alongside the html and whenever Hackage renders the index page for a package it would take the module list and description from that file.

It’s not hard to implement, we already have everything necessary at hand.
